repos/system_upgrade/el7toel8/libraries/rhsm.py

```diff
@@ -108,26 +113,75 @@ def get_attached_skus(context, rhsm_info):
             rhsm_info.attached_skus = _RE_SKU_CONSUMED.findall(result['stdout'])
 
 
-@with_rhsm
 def get_available_repo_uids(context, rhsm_info):
+    """Deprecated. Repositories have ids, not uids. Use get_available_repo_ids instead."""
+    return get_available_repo_ids(context, rhsm_info)
+
+
+def get_available_repo_ids(context, rhsm_info):
     """
-    Retrieves all available repositories names from the subscription manager.
+    Retrieve repo ids of all the repositories available through the subscription-manager.
 
     :param context: An instance of a mounting.IsolatedActions class
     :type context: mounting.IsolatedActions class
     :param rhsm_info: An instance of a RHSMInfo derived model.
     :type rhsm_info: RHSMInfo derived model
     """
     if not rhsm_info.available_repos:
-        with _handle_rhsm_exceptions():
-            result = context.call(['subscription-manager', 'repos'], split=False)
-            rhsm_info.available_repos = _RE_REPO_UID.findall(result['stdout'])
+        result = context.call(['yum', 'repoinfo'])
+        all_repos = list(get_repos(result['stdout']))
+        rhsm_info.available_repos = [repo.id for repo in all_repos if repo.file == '/etc/yum.repos.d/redhat.repo']
+
+
+def get_repos(repos_raw):
+    """
+    Generator providing all the repos available through yum/dnf.
+
+    :rtype: Iterator[:py:class:`leapp.libraries.common.rhsm.Repo`]
+    """
+    # TODO: Log a warning on the following msg that `yum repoinfo` may produce:
+    #  "Repository rhel-7-server-eus-rpms is listed more than once in the configuration"
```

Comment:
@bocekm shouldn't we report that as well? I mean in the report. As it is something what I believe it should not happen on correctly configured system and it could be nice if user resolve that on their own. As it looks like they produce the repo file with the id, that should not appear on the rhel-8 system. as well, if there could be set `enabled=1` - as it's kinda custom repository once it is provided out of the repo file managed by rhsm - so it could happen, they will be able to "install" some rpms that are not supposed to be delivered on rhel-8.

Reply:
of course, that could be part of different PR in future. It doesn't have to be resolved in this one.

Reply:
I'm detecting it and inhibit the upgrade now. I just hope that I can report an inhibitor from a common library.